helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -393,9 +393,6 @@ public AccessResult doSet(String path, T record, int expectVersion, int options)
           result._retCode = RetCode.ERROR;
           return result;
         }
-      } catch (ZkBadVersionException e) {
-        LOG.debug("Exception while setting path: " + path, e);
-        throw e;
```

Comment:
we may introduce a behavior change here. As now when we get `ZkBadVersionException`, it will go to general exception branch.